src/binarySearch.js

```diff
@@ -0,0 +1,30 @@
+export default function binarySearch( arr, query){
+  if( arr == null ){
+    return 'Cannot search null'
+  }
+  if( arr.constructor !== Array ){
+    return 'Please use Arrays'
+  }
+
+  const splitSearch = (start, end) => {
+    var midpoint = Math.round( (end-start)/2 + start)
+
+    if( query === arr[midpoint]){
+      return midpoint
+    }
+
+    if( query > arr[midpoint] ){
+      if( start === end ){
```

Comment:
Correction:  We moved this check up so it comes before the if statements with the recursive calls and it works there!